src/app/toggle-scroll.service.ts

```diff
@@ -0,0 +1,12 @@
+import { Inject, Injectable } from '@angular/core';
+import { DOCUMENT } from '@angular/platform-browser';
+
+@Injectable()
+export class ToggleScrollService {
+  set allowScroll(val: boolean) {
+    document.body.style.overflowY = val ? null : 'hidden';
```

Comment:
Calling document directly works because it is a global var, but if we're accessing the `document` through injection should this be `this.document.body.style.overflowY`?